### DISCUSSION TITLE
the second it block is redirecting to the loginpage needed help

### DISCUSSION THREAD

**Discussion by nrcsk** [cypress-io/cypress#16894] [Questions and Help]
```js
/// <reference types="Cypress" />

import LoginPage from '../PageObjectsDDI/LoginPage'
import ClientsPage from '../PageObjectsDDI/ClientsPage'
describe('Clients Module' , function(){
    before(function(){

        cy.fixture('testData').then(function(data)
        {
               this.data=data
        })
        
    })
    it('Login DDI' , function(){

        const lp=new LoginPage()
        lp.visit()
        lp.fillemail(this.data.email)
        lp.fillpassword(this.data.password)
        lp.submit()
        cy.get('#app > main > div > div > div.page-header > div.header-contents > h1').and('contain','Onboarding & Maintenance')
        })
        it('Clients',function(){
            const cp=new ClientsPage()
            cp.Clients()
            cy.get('#app > main > div > div > div.page-header > div.header-contents > h1').and('contain','Clients')






        })






})
```

**Comment by sandra-ouadghiri** [comment]
@chandunasaka > On a general note, please submit your code with no custom functions so we can help. And only native Cypress, because we don't know what you put in those classes. Also, making this code indented and proper for us to review will aise the chance you getting a reply.

If I was to bet, I would say : it() indicates isolated scenarios. Not part 1 then part 2. And you don't seem to have followed that.

Cypress does us a favor and purges everything between 2 scenarios to prevent cross-scenario contamination. By purging cookies and local storage it often purges login sessions. Then your product would probably redirect you to a login pasge as a consequence.

If you need the content of it() n°1 to work, then the end of your code should not be in it() n°2 but still in it() n°1. As a rule, it() should work whatever their order of execution is. So do you thing that an execution order with it() n°2 first and then it() n°1 would work? If no, then you have an issue.

If you want to divide into 2 scenario, then you need to make them independant, by managing login each time, for ex via an api login or a auth header.

On a side note:
- Cypress has a great doc about good practices that I highly recommend
- I would not advice PageObject for the sake of it. Is ip.visit() really better code than cy.visit()? Will really be sustenance easier when more code is involved? Cypress manages a lot of things out-of-the-box that justified heavy pageObject in Selenium. As a general advice, try simple working code first. Then if required, you can still create functions/class/whatever out of blocks that are repeated in your code. But later, make code work first, do architecture stuff then.
- I would advise to work on those selectors. That's so fragile to navigate your DOM like this. Maybe you can get an attribute that would be identifying your h1 blocks. If you struggle, a QA senior or a front dev may help.
- Also, testing text is a door to insane sustenance as not i18n-resiliant. So maybe not a problem for an EN-only product or for some precise strings such as a company name. Still, be aware of that.

Those are just additional friendly advises that hopefully will help. If you disagree, feel free to ignore. Take care and good luck!

**Reply by nrcsk** [reply]
This is the code @sandra-ouadghiri please review and come up with a solution thanks

**Reply by sandra-ouadghiri** [reply]
@chandunasaka > First: Does your code work if all 3 'scenarios' are into the same it() (one it not 3)?

**Reply by nrcsk** [reply]
Yes absolutely @sandra-ouadghiri
